core/src/main/scala/no/nrk/bigquery/BQTableLike.scala

```diff
@@ -90,6 +93,28 @@ case class BQTableRef[+P](
 
   override def withTableType[PP](tpe: BQPartitionType[PP]): BQTableRef[PP] =
     BQTableRef(tableId, tpe)
+
+  def asFragment: BQSqlFrag = tableId.asFragment
+}
+
+case class BQAppliedTableValuedFunction[+P, N <: Nat](
+    tvf: TVF[P, N],
+    args: Sized[IndexedSeq[BQSqlFrag], N]
+) extends BQTableLike[P] {
+  override def tableId: BQTableId =
+    BQTableId(tvf.name.dataset, tvf.name.name.value)
+
+  override def partitionType: BQPartitionType[P] = tvf.partitionType
+
+  override def unpartitioned: BQTableLike[Unit] =
+    withTableType(BQPartitionType.ignoredPartitioning(partitionType))
+
+  override def withTableType[PP](tpe: BQPartitionType[PP]): BQTableLike[PP] =
+    BQAppliedTableValuedFunction(
+      TVF(tvf.name, tpe, tvf.params, tvf.query, tvf.schema, tvf.description),
```

Comment:
maybe TVF should have a withPartitionType here to help with doing this correctly.

Reply:
I've normalized the values from TVF to BQAppliedTableValuedFunction so this comment isn't applicable anymore.